src/lowlevel.ts

```diff
@@ -0,0 +1,109 @@
+import {
+  type Config,
+  extractBaseUrl,
+  extractSigner,
+  type Signal,
+  type ReadConfig,
+} from "./helpers/index.js";
+import { prepareCreateTable, createTable } from "./registry/create.js";
+import { prepareRunSQL, runSQL } from "./registry/run.js";
+import {
+  type ExtractedStatement,
+  type WaitableTransactionReceipt,
+  wrapTransaction,
+} from "./registry/utils.js";
+import {
+  type ObjectsFormat,
+  type ValueOf,
+  getQuery,
+} from "./validator/query.js";
+import { ApiError } from "./validator/index.js";
+
+export async function exec(
+  config: Config,
+  { type, sql, tables: [first] }: ExtractedStatement
+): Promise<WaitableTransactionReceipt> {
+  const signer = await extractSigner(config);
+  const chainId = await signer.getChainId();
+  const baseUrl = await extractBaseUrl(config, chainId);
+  const _config = { baseUrl, signer };
+  const _params = { chainId, first, statement: sql };
+  switch (type) {
+    case "create": {
+      const { prefix, ...prepared } = await prepareCreateTable(_params);
+      const tx = await createTable(_config, prepared);
+      return await wrapTransaction(_config, prefix, tx);
+    }
+    /* c8 ignore next */
+    case "acl":
+    case "write": {
+      const { prefix, ...prepared } = await prepareRunSQL(_params);
+      const tx = await runSQL(_config, prepared);
+      return await wrapTransaction(_config, prefix, tx);
+    }
+    /* c8 ignore next 2 */
+    default:
+      throw new Error("invalid statement type: read");
+  }
+}
+
+function catchNotFound(err: unknown): [] {
+  if (err instanceof ApiError && err.status === 404) {
+    return [];
+  }
+  throw err;
+}
+
+export async function queryRaw<T = unknown>(
+  config: ReadConfig,
+  statement: string,
+  opts: Signal = {}
+): Promise<Array<ValueOf<T>>> {
```

Comment:
Note the amazing typed responses we have here. This type is smart enough to _know_ that a raw response will basically unwrap a type, and just return its "values". Cool!